### PR TITLE
[DW-170] Adding external resource field to legacy pub

### PR DESCRIPTION
**Pull request by giacomolm**

### CODE REVIEW

File: repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/legacy-publication.ftl [hunk 1]
```diff
@@ -134,7 +134,8 @@
                 <div class="panel panel--grey push-half--bottom">
                     <h3><@fmt.message key="headers.resources"/></h3>
                     <ul data-uipath="ps.publication.resources">
-                        <#list legacyPublication.attachments as attachment>
+                        ${legacyPublication.getExtAttachments()?size}
```
Comment (wojtek-oledzki) [inline repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/legacy-publication.ftl]: Do we need that in the template?

File: repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/legacy-publication.ftl [hunk 2]
```diff
@@ -134,7 +134,8 @@
...
+                        <#list legacyPublication.getExtAttachments() as attachment>
                             <li class="attachment">
                                 <a title="${attachment.text}" href="<@hst.link hippobean=attachment.resource/>" onClick="logGoogleAnalyticsEvent('Download attachment','Publication','${attachment.resource.filename}');">${attachment.text}</a>;
```
Comment (wojtek-oledzki) [inline repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/legacy-publication.ftl]: take a look at "external-resource-demo" branch.

```
<@externalstorageLink attachment.resource; url>
    <a title="${attachment.text}" href="${url}" onClick="logGoogleAnalyticsEvent('Download attachment','Publication','${attachment.resource.filename}');">${attachment.text}</a>;
</@externalstorageLink>
```

File: repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/legacy-publication.ftl (hunk 1 above)
Comment (giacomolm) [reply repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/legacy-publication.ftl]: @robValtech could you have a look at this?

File: repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/legacy-publication.ftl (hunk 2 above)
Comment (giacomolm) [reply repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/legacy-publication.ftl]: @robValtech could you have a look at this?

**Comment by giacomolm** [comment]
Made some changes (removed a leftover) and discussed with @robValtech, he will dive into the macro that has been proposed

File: repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/legacy-publication.ftl (hunk 2 above)
Comment (andyleitch) [reply repository-data/webfiles/src/main/resources/site/freemarker/publicationsystem/legacy-publication.ftl]: We'll make the change mentioned above after it is merged to master and has been fully tested etc. Need to progress with this as is for now - blocking our migration

**Pull request merged**